test/config/type/function-with-argv/function-with-argv.test.js

```diff
@@ -0,0 +1,14 @@
+'use strict';
+const { existsSync } = require('fs');
+const { resolve } = require('path');
+const { run } = require('../../../utils/test-utils');
+
+describe('function configuration', () => {
+    it('is able to understand a configuration file as a function', () => {
+        const { stderr, stdout } = run(__dirname, ['--mode', 'development']);
+        expect(stderr).toBeFalsy();
+        expect(stdout).toBeTruthy();
+        // Should generate the appropriate files
```

Comment:
@evilebottnawi any way to make these tests better?

Reply:
is it possible to log `argv` and check its value? It should not be `undefined`

Reply:
I think testing is enough in this case

Reply:
oops I added a commit for checking argv too, I guess it's fine